clang/lib/StaticAnalyzer/Checkers/ArrayBoundCheckerV2.cpp

```diff
@@ -33,7 +33,66 @@ using namespace taint;
 using llvm::formatv;
 
 namespace {
-enum OOB_Kind { OOB_Precedes, OOB_Exceeds, OOB_Taint };
+class StateUpdateReporter {
+  const SubRegion *Reg;
+  NonLoc ByteOffsetVal;
+  std::optional<QualType> ElementType = std::nullopt;
+  std::optional<int64_t> ElementSize = std::nullopt;
+  bool AssumedNonNegative = false;
```

Comment:
Maybe it's only me, but I'd prefer to put the bool flag at the end to have better storage layout (less padding bytes).
As a rule of thumb, order descending by alignment.

Reply:
I placed them in this order because this "lower bound, then upper bound" order is the natural way to display them; but I can flip them if you think that this memory footprint reduction is significant.